common/src/main/java/io/pravega/common/io/filesystem/FileOperations.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.io.filesystem;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+
+import java.io.File;
+import java.util.HashMap;
+
+@Slf4j
+public class FileOperations {
+
+    public static boolean cleanupDirectories(HashMap<?, File> toDelete) {
+        for (File dir : toDelete.values()) {
```

Comment:
This change is due to the unfortunate reality of pleasing code coverage. A way was required to (platform independently) test for a failure scenario. The previous solution had several problems which made this difficult:

1. The visibility of `cleanupDirectories` and other `BookKeeperServiceRunner` fields made it hard to unit test `cleanupDirectories` at all in the first place.
2. Unit testing `cleanupDirectories` within any other the `BookKeeper*` tests seems very forced and unnatural.
3. Forcing an `IOException` to be thrown (across platforms) is also quite difficult -- it is much easier to force an `IllegalArgumentException`.
4. We still need some way to determine whether or not an exception was thrown. Previously the exception was only caught and logged. 

At the expensive of creating two small classes, this avoids the above problems.